### DISCUSSION TITLE
File Upload in a Form: How to Disable and Enable 'Save' Buttons?

### DISCUSSION THREAD

**Discussion by aytackayin** [filamentphp/filament#10023] [Help]
In a form, how can a custom button be disabled when start file uploading and enabled when it finishes? Below, I want to disable the 'save' and 'saveAnother' buttons until the file upload process is completed. Then, I want them to become enabled.

File Upload Component in form PostResource.php
```
FileUpload::make('attachments')
    ->translateLabel()
    ->multiple()
    ->disk('public')
    ->directory('category')
    ->imageEditor()
    ->visibility('private')
    ->reorderable()
    ->appendFiles()
    ->preserveFilenames(false)
    ->getUploadedFileNameForStorageUsing(
        fn (TemporaryUploadedFile $file): string => (string) str($file->getClientOriginalName())
            ->prepend('category-'),
    )
    ->openable(),
```
Custom button in CreatePost.php
```
protected function getFormActions(): array
{
    return [
        Action::make('save')
            ->label(__('button.create'))
            ->action('saveGo'),
        Action::make('saveAnother')
            ->label(__('button.create_another'))
            ->color('success')
            ->action('saveAnother'),
        $this->getCancelFormAction()
            ->label(__('button.cancel'))
            ->keyBindings(['esc'])
            ->url($this->previousUrl),
    ];
}
```

**Comment by mauricoder** [comment]
I have the same problem. As anxious users who press the save button many times while file is uploading will cause a 500 error. Is this a solution or where you having problems with this implementation?

Thanks!

**Reply by leandrocfe** [reply]
Are you using the Panel builder or Form builder only?

**Reply by mauricoder** [reply]
The Panel Builder. Also I'm not using the FileUpload, but the SpatieMediaLibraryFileUpload. Thanks for the help!

**Comment by edwink75** [comment]
Im facing the same issue, that I'd like to disable the submit button until all images are uploaded. Is there meanwhile a solution?

The ->afterStateUpdated is triggered after each image is uploaded, so I can't use that. Unless I can count the images that are still in the selection to be uploaded? 

Are there browser events I can work with? One for loading the Fileupload field and one after all files are uploaded? I'm surprises there are so few people having the same issue.

Laravel	10.49.0
Filament	3.2.137
Livewire	3.5.12
PHP	8.3.25

I will keep digging a bit further, maybe there is a way to use the alpineJS of the field?

**Comment by Estigy** [comment]
There's already a second discussion for this very same topic:
https://github.com/filamentphp/filament/discussions/211
